### PR TITLE
add chem factory related cargo bounty

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
add: pill chemfactory bounty 
/:cl:

bounty of 80-140 pills of not easy chems of 30-50 units
gives chemists more things to do with high productivity

### CODE REVIEW

**Comment by carlarctg** [comment]
yikes does it really take that much effort to type a description?

this seems very, very easy to game if you're trying to do chemfactory bounties as all of these are easily makeable in a dispenser but i guess that's what you get with the added complexity it brings
maybe make it 20 to 30 pills instead?

**Comment by necromanceranne** [comment]
Wait am I reading this right? You can send a bunch of 1 units pills for the bounty? Hmmm

**Comment by TheVekter** [comment]
https://tgstation13.org/wiki/Getting_Your_Pull_Accepted

Also, as others have noted, this seems easy to abuse. What's the purpose of this? To make it so the bounty recognizes pills and not just units of drugs?

**Comment by Tlaltecuhtli** [comment]
> Wait am I reading this right? You can send a bunch of 1 units pills for the bounty? Hmmm

no you need a minimum wanted_vol which is finalized on new()

> this seems very, very easy to game if you're trying to do chemfactory bounties as all of these are easily makeable in a dispenser but i guess that's what you get with the added complexity it brings

increased value to minimum 80 pills of 30u+  hard chems 😎  good luck making them in chem dispenser

**Comment by TheChosenEvilOne** [comment]
Reward should be calculated with amount and vol, maybe even give each reagent a base value. Also you can get a bounty of 1 pill with 1u for 15000 credits which is ridiculous.

**Review by Tlaltecuhtli — COMMENTED** [review]
minimum value
**var/required_ammount = 80**
	var/shipped_ammount = 0
	var/datum/reagent/wanted_reagent
	**var/wanted_vol = 30**

File: code/modules/cargo/bounties/reagent.dm [hunk 1]
```diff
@@ -188,3 +188,67 @@
...
+/datum/bounty/pill/completion_string()
+	return {"[shipped_ammount]/[required_ammount] pills"}
+
+/datum/bounty/pill/can_claim()
+	return ..() && shipped_ammount >= required_ammount
+
+/datum/bounty/pill/applies_to(obj/O)
+	if(!istype(O, /obj/item/reagent_containers/pill))
+		return FALSE
+	if(O.reagents && (O.reagents.get_reagent_amount(wanted_reagent.type) >= wanted_vol))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/cargo/bounties/reagent.dm]:
```suggestion
	if(O?.reagents.get_reagent_amount(wanted_reagent.type) >= wanted_vol))
```

File: code/modules/cargo/bounties/reagent.dm [hunk 2]
```diff
@@ -188,3 +188,67 @@
 	name = wanted_reagent.name
 	description = "CentCom is paying premium for the chemical [name]. Ship a container of it to be rewarded."
 	reward += rand(0, 5) * 750 //6000 to 9750 credits
+
+/datum/bounty/pill
+	var/required_ammount = 80
+	var/shipped_ammount = 0
+	var/datum/reagent/wanted_reagent
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/cargo/bounties/reagent.dm]: this is a type not an instance

File: code/modules/cargo/bounties/reagent.dm [hunk 3]
```diff
@@ -188,3 +188,67 @@
...
+		/datum/reagent/drug/crank,\
+		/datum/reagent/nitrous_oxide,\
+		/datum/reagent/barbers_aid,\
+		/datum/reagent/pax,\
+		/datum/reagent/flash_powder,\
+		/datum/reagent/phlogiston,\
+		/datum/reagent/firefighting_foam)
+
+	var/reagent_type = pick(possible_reagents)
+	wanted_reagent = new reagent_type
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/cargo/bounties/reagent.dm]: why do you make this an instance? just make it a type man...

**Comment by ExcessiveUseOfCobblestone** [comment]
use the template homeslice

File: code/modules/cargo/bounties/reagent.dm (hunk 3 above)
Comment (Tlaltecuhtli) [reply code/modules/cargo/bounties/reagent.dm]: .type doesnt work

**Comment by ExcessiveUseOfCobblestone** [comment]
i was referring to the template for New PRs but that works too lol

File: code/modules/cargo/bounties/reagent.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/cargo/bounties/reagent.dm]: it does if you do it right

**Pull request merged**